setup-ssh/src/ec2-utils.ts

```diff
@@ -1,18 +1,17 @@
 import {HttpClient} from '@actions/http-client'
+import {IHttpClientResponse} from '@actions/http-client/interfaces'
 
 export async function getEC2Metadata(category: string): Promise<string> {
   const maxRetries = 10
   const http = new HttpClient('seemethere/add-github-ssh-key', undefined, {
     allowRetries: true,
     maxRetries
   })
-  // convert these two curls:
-  // curl -H "X-aws-ec2-metadata-token: $(curl -s -X PUT "http://169.254.169.254/latest/api/token" -H "X-aws-ec2-metadata-token-ttl-seconds: 30")" -fsSL "http://169.254.169.254/latest/meta-data/${category}"
-    const tokenResponse = await http.put(
-    `http://169.254.169.254/latest/api/token`, undefined, {
-      headers: {
-        'X-aws-ec2-metadata-token-ttl-seconds': '30'
-      }
+  const tokenResponse: IHttpClientResponse = await http.put(
+    `http://169.254.169.254/latest/api/token`,
+    '',
+    {
+      'X-aws-ec2-metadata-token-ttl-seconds': '30'
     }
   )
 
```

Comment:
Don't you want to do some error-checking here, similar to `resp.message.statusCode` on line 31?

Reply:
Sorry, what I've meant that you need to add something like
```typescript
if (tokenResponse.message.statusCode !== 200) {
    return ''
  }
```

Otherwise, your `with-ssh` action will throw an exception on line 24, because `tokenResponse.readBody()` throws an exception if statusCode is anything but 200

Reply:
yes, that exact code is already on line 19 and 20 (unchanged in this commit) ;)